src/axom/sidre/core/DataStore.hpp

```diff
@@ -191,6 +187,29 @@ class DataStore
 
   //@}
 
+  //@{
+  //!  @name Accesors for buffer collections.
+  //!
+  //! These methods can be used to iterate on the collection of buffers
+  //! Example:
+  //!      for (auto& buffer : ds->buffers())
+  //!      {
+  //!          /// code here using buffers
+  //!      }
+
+  /*!
+   * \brief Returns a reference to the collection of buffers
+   */
+  BufferCollection& buffers() { return *m_buffer_coll; }
```

Comment:
Do we really want to expose a non-const reference to the buffer collection?

Reply:
I added an `iterator_adaptor` utility class to `ItemCollection` to only expose the functions required for iteration, e.g. `begin()` and `end()`